src/Plugins/Nop.Plugin.Product.Backup/Factory/ProductBackupFactory.cs

```diff
@@ -0,0 +1,146 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Nop.Core;
+using Nop.Plugin.Product.Backup.Models;
+using Nop.Plugin.Product.Backup.Models.Settings;
+using Nop.Plugin.Product.Backup.Services.Picture;
+using Nop.Plugin.Product.Backup.Services.Product;
+using Nop.Services.Configuration;
+using Nop.Services.Media;
+using Nop.Web.Areas.Admin.Infrastructure.Mapper.Extensions;
+
+namespace Nop.Plugin.Product.Backup.Factory;
+
+public class ProductBackupFactory : IProductBackupFactory
```

Comment:
decouple settings part of the product model creation in 2 different classes/interfaces